### PR TITLE
[ICDS] month filter in SSD report start from Feb 2018

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @calellowitz @Rohit25negi 

I changed the month filter for SDD report and now we can choose only dates after Feb 2018. Also, I added info message when the user loading the report and have chosen date before Feb 2018.

Need QA: no
Environment: ICDS
Locally tested: yes

Regards,
Łukasz

### CODE REVIEW

**Review by calellowitz — COMMENTED** [review]
Is there a ticket this is from? The code looks fine but I am not super happy with redirecting the user to the current month. That seems quite surprising to the user. I am hoping to figure out the rationale for that and if there might be a better option. Can you also include a screenshot of the warning message?

**Comment by lwyszomi** [comment]
Hi @calellowitz Sorry, I forgot to add ticket: https://dimagi-dev.atlassian.net/browse/ICDS-644

Please don't merge this I found one case where this not working. I will update the PR as soon as possible.

**Comment by lwyszomi** [comment]
Hi @calellowitz I updated the PR, now all work correctly. Also, I'm adding the screenshot:
info_message_sdd

File: custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ function MonthModalController($location, $uibModalInstance) {
     vm.years = [];
     vm.monthsCopy = [];
 
+    var startDate = $location.path().indexOf('service_delivery_dashboard') === -1 ? 2017 : 2018;
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js]: This should be starting from feb 2019 not feb -2018 as ticket says.

File: custom/icds_reports/templates/icds_reports/icds_app/service-delivery-dashboard.html [hunk 2]
```diff
@@ -23,6 +23,9 @@
         </div>
     </div>
     <info-message></info-message>
+    <div class="alert alert-info" ng-show="$ctrl.showMessage">
+      <strong>Info!</strong> The date was changed to the <strong>current month</strong> because the report was loaded for a period which is not supported. Supported period is after Feb 2018.
```
Comment (Rohit25negi) [inline custom/icds_reports/templates/icds_reports/icds_app/service-delivery-dashboard.html]: Feb 2019

File: custom/icds_reports/static/js/directives/navigation/navigation.directive.js [hunk 3]
```diff
@@ -26,10 +26,25 @@ function NavigationController($window, $scope, $route, $routeParams, $location,
...
+        var selectedMonth = $location.search()['month'] !== void(0) ? parseInt($location.search()['month']) : new Date().getMonth() + 1;
+        var selectedYear = $location.search()['year'] !== void(0) ? parseInt($location.search()['year']) : new Date().getFullYear();
+
+        var isServiceDeliveryDashboard = path.indexOf('service_delivery_dashboard') !== -1;
+        if (isServiceDeliveryDashboard && selectedYear <= 2018) {
+            if (selectedYear < 2018 || (selectedYear === 2018 && selectedMonth === 1)) {
+                $location.search('month', new Date().getMonth() + 1);
+                $location.search('year', new Date().getFullYear());
+                $rootScope.dateChanged = true;
+                params = $location.search();
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/navigation/navigation.directive.js]: Will it change the query param in the url also? or its just a ajax request?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left Some comments to correct the filter. and one general question.

File: custom/icds_reports/static/js/directives/navigation/navigation.directive.js (hunk 3 above)
Comment (lwyszomi) [reply custom/icds_reports/static/js/directives/navigation/navigation.directive.js]: The params will be changed in both locations.

**Review by Rohit25negi — APPROVED** [review]
Looks good to me. Good to merge once the test cases passes

**Comment by calellowitz** [comment]
Made the same note in teh revert PR, but just to make sure it is seen, I reverted this because the ticket says to prompt the user to select a month, not redirect to the current month. I agree with the ticket that prompting the user to select is a better UI so am reverting this change until that is implemented

**Pull request merged**